src/main/java/io/fabric8/maven/docker/service/RunService.java

```diff
@@ -464,11 +467,23 @@ private void startContainer(ImageConfiguration imageConfig, String id, GavLabel
     }
 
     private void updateMappedPortsAndAddresses(String containerId, PortMapping mappedPorts) throws DockerAccessException {
-        Container container = queryService.getMandatoryContainer(containerId);
-        if (container.isRunning()) {
-            mappedPorts.updateProperties(container.getPortBindings());
-        } else {
-            log.warn("Container %s is not running anymore, can not extract dynamic ports",containerId);
+        for (int attempt = 0; attempt < 5; attempt++) {
```

Comment:
Hello, Do you think we should add `Thread.currentThread().isInterrupted()` in this loop terminating condition here?

Reply:
Good shout. Added